### PR TITLE
Concurrency lib fixes

### PR DESCRIPTION
**Pull request by Effectualness**

### CODE REVIEW

File: concurrency/utils.c [hunk 1]
```diff
@@ -1,43 +1,66 @@
...
 
+//Used to manage sem value changes, and for error handler
+CSSem *vizconSem;
+
+//Used to create a semaphore that manages sem value changes, and errors
+void vizconSemCheck()
+{
+    if(vizconSem != NULL)
+        semClose(vizconSem);
+        vizconSem = NULL;
```
Comment (TheIceCreamBear) [inline concurrency/utils.c:17]: What does this `if` need to cover? are there missing braces?

**Review by TheIceCreamBear — COMMENTED** [review]
Looks good, one comment, and im also going to add one commit

File: examples/bank.c [hunk 2]
```diff
@@ -46,14 +46,14 @@ void* Teller(void* param)
 {
     int accountNum, amount, teller = (int)param;
     srand(vcThreadId());
```
Comment (TheIceCreamBear) [inline examples/bank.c:48]: is srand thread specific?

File: examples/bank.c [hunk 3]
```diff
@@ -75,7 +75,7 @@ void* MainBranch(void* param)
 {
     int accountNum, amount;
     srand(vcThreadId());
```
Comment (TheIceCreamBear) [inline examples/bank.c:77]: is srand thread specific (duplicate question)

**Review by TheIceCreamBear — COMMENTED** [review]
Left some other comments, and be careful about VLAs

File: concurrency/utils.c (hunk 1 above)
Comment (Effectualness) [reply concurrency/utils.c]: There was missing braces, this is now fixed

**Pull request merged**